library/pod-security-policy/capabilities/src.rego

```diff
@@ -0,0 +1,52 @@
+package capabilities
+
+violation[{"msg": msg}] {
+  container := input.review.object.spec.containers[_]
+  has_disallowed_capabilities(container)
+  msg := sprintf("container <%v> has a disallowed capability. Allowed capabilities are %v", [container.name, get_default(input.parameters, "allowedCapabilities", "NONE")])
+}
+
+violation[{"msg": msg}] {
+  container := input.review.object.spec.containers[_]
+  missing_drop_capabilities(container)
+  msg := sprintf("container <%v> is not dropping all required capabilities. Container must drop all of %v", [container.name, input.parameters.requiredDropCapabilities])
+}
+
+
+
+violation[{"msg": msg}] {
+  container := input.review.object.spec.initContainers[_]
```

Comment:
Is there a reason to explicitly call out the init containers or can we find all containers and initcontainers in one call?

Reply:
It decouples the two cases to make testing easier. It also allows us to say which field the container is in, making the message more informative.

Reply:
none of the other policies do this. is it sufficient to return container name?

Reply:
It seems less useful. TBH if consistency is the issue it's better to uplift the others later than to make the current constraint less useful.

Also, I'm not sure what benefit we get in trying to be exactly uniform across different constraints.

Reply:
I'm ok either way. But why isnt container.name good enough?

Reply:
It's not that it's not good enough, but container + what field it lives under == better as it lets the user know where to look and potentially disambiguates if they're using the same name twice.

Reply:
Looks like you can't name `initContainers` and `containers` with same name. 
```
The Pod "myapp-pod" is invalid: spec.initContainers[0].name: Duplicate value: "myapp-container"
```
I think this will make regos longer without extra information.

Reply:
Which field it lives under is still extra information.

Reply:
Also is it regos or errors your interested in? For the Rego src, the extra benefit is reduced brittleness, problems with checking initContainers do not interfere with Containers and vice-versa.